subdomain-replace.go

```diff
@@ -0,0 +1,113 @@
+package rdns
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+// Replace is a resolver that modifies queries according to regular expressions
+// and forwards the modified queries to another resolver. Responses are then
+// mapped back to the original query string.
+type SubDomainReplace struct {
+	id       string
+	resolver Resolver
+	exp      subDomainReplaceExpressions
+}
+
+var _ Resolver = &SubDomainReplace{}
+
+type subDomainReplaceExp struct {
+	from string
+	to   string
+}
+
+type subDomainReplaceExpressions []subDomainReplaceExp
+
+func (r subDomainReplaceExp) substitute(qname string) (string, bool) {
+	fromDomain := r.from
+	toDomain := r.to
+	// from {}.ctptech.dev to {}.charlesp.tech
+	// from test.ctptech.dev to test.charlesp.tech
+	if strings.HasSuffix(qname, fromDomain) {
+		str := strings.Replace(qname, fromDomain, toDomain, 1)
```

Comment:
I think there's a small issue with just doing a simple replace since the "fromDomain" string can show up more than once in the domain. Let's say you have `fromDomain = .local.` and the domain name being looked up is `example.local.something.local.` then it'd replace the first occurrance. It might be best to simply strip the `fromDomain` with `strings.TrimSuffix()`, then add the `toDomain` value to the result with `+` like so:
```go
sub := strings.TrimSuffix(qname, fromDomain)
str := sub+toDomain
return str, true
```